XenAdmin/Wizards/PatchingWizard/AutomatedUpdatesBasePage.cs

```diff
@@ -189,30 +189,33 @@ private void StartNewWorker(string poolName, List<HostPlan> planActions, List<Pl
 
         private void WorkerProgressChanged(object sender, ProgressChangedEventArgs e)
         {
-            var actionsWorker = sender as UpdateProgressBackgroundWorker;
-            if (actionsWorker == null)
+            var bgw = sender as UpdateProgressBackgroundWorker;
+            if (bgw == null)
                 return;
 
-            if (!actionsWorker.CancellationPending)
+            if (!bgw.CancellationPending)
             {
                 PlanAction action = (PlanAction)e.UserState;
                 if (action != null)
                 {
                     if (!action.IsComplete)
                     {
-                        if (!actionsWorker.InProgressActions.Contains(action))
-                            actionsWorker.InProgressActions.Add(action);
+                        if (!bgw.InProgressActions.Contains(action))
+                            bgw.InProgressActions.Add(action);
                     }
                     else
                     {
-                        if (!actionsWorker.DoneActions.Contains(action))
-                            actionsWorker.DoneActions.Add(action);
-                        actionsWorker.InProgressActions.Remove(action);
+                        if (!bgw.DoneActions.Contains(action))
+                            bgw.DoneActions.Add(action);
+                        bgw.InProgressActions.Remove(action);
+
+                        if (!failedWorkers.Contains(bgw))

```

Comment:
I think we should only add the worker to the failedWorkers if action.Error is not null (so the worker had failed)

Reply:
oops! 😳